resources/views/components/input/text.blade.php

```diff
@@ -1,12 +1,63 @@
-<x-input.input
-    :model="$model ?? null"
-    :type="$type ?? 'text'"
-    :attribute="$attribute ?? 'text'"
-    :icon="$icon ?? false"
-    :inline="$inline ?? false"
-    :placeholder="$placeholder ?? false"
-    :required="$required ?? false"
-    :disabled="$disabled ?? false"
-    :readonly="$readonly ?? false"
-    :help="$help ?? false"
-/>
+<?php
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+?>
+
+@props([
+    'disabled' => false,
+    'fullWidth' => true,
+    'help' => null,
+    'icon' => false,
+    'id' => null,
+    'inline' => false,
+    'label' => true,
+    'maxlength' => 255,
+    'model' => null,
+    'name' => null,
+    'placeholder' => false,
+    'readonly' => false,
+    'required' => false,
+    'requiredSilent' => false,
+    'type' => 'text',
+    'value' => '',
+])
+
+<?php
+if ($model && !$model instanceof Model) {
+    throw new Exception('"model" is not an Eloquent model');
+}
+
+$id = $id ?: 'input_' . Str::random();
+?>
+
+<x-form-field
+    :help="$help"
+    :hidden="$type === 'hidden'"
+    :icon="$icon"
+    :id="$id"
+    :inline="$inline"
+    :model="$model"
+    :name="$name"
+>
+    @if($label && $type !== 'hidden')
```

Comment:
It seems like `$type === 'hidden'` should set "sr-only" rather than removing the label from the DOM.

Reply:
I assumed hidden inputs to not require labels 🤔

Reply:
Apologies, I actually misinterpreted in my original review what `$type` was doing here. You're right, input fields with type="hidden" do not require labels.